### PR TITLE
Porting Cargo small-order support, cargo goodies packs, beating vendors for loot/squish.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Porting tgstation PRs #43402 (by zxaber), #47799 and #51216 (by Ryll-Ryll), plus a little offtopic fix for ghosts and wires by AnturK.

Done a few tweaks so the newer goodies feature can co-exist with the older small orders one, without much weapons stuff atm (eh, could throw in tackler gloves or something later).
Also made a good chunk of material packs small orders and split the carpet mega packs into singular ones.

Also removed unused bits of the old embedded behavior.

## Why It's Good For The Game
Small packs that are actually affordable by the average jobbie.
The coupon feature does need that beat-em-vendor-up-and-get-squished update to work correctly btb.

## Changelog
:cl: Ghommie (porting PRs by zxaber, Ryll-Ryll, AnturK)
tweak: Certain small items purchased through cargo now get grouped into a single box. They also are immune to the 10% private account fee.
add: Added single-order options for several existing products in the Cargo Catalog.
tweak:  Medkit listings are now single-pack items, and considered small items that get grouped into single boxes. Price for medkits is as close to unchanged as is reasonable.
add: You can now beat on vending machines to try and knock loose free stuff! You can also almost kill yourself doing it, so it's your call if your life is worth ten bucks.
add: Cigarette packets now have coupons on the back for small cargo items! Smoking DOES pay!
tweak: Some single/small items in cargo have been rebranded as goodies, come in lockboxes rather than crates, and can only be purchased with private accounts.
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Im really against small orders. As its cheaper to buy them 1 at a time

File: code/modules/cargo/coupon.dm [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+#define COUPON_OMEN "omen"
+
+obj/item/coupon
```
Comment (Trilbyspaceclone) [inline code/modules/cargo/coupon.dm]:
```suggestion
/obj/item/coupon
```

File: code/modules/cargo/order.dm [hunk 2]
```diff
@@ -57,58 +60,64 @@
...
-			M.errors &= ~MANIFEST_ERROR_ITEM
-		else
-			var/lost = max(round(C.contents.len / 10), 1)
-			while(--lost >= 0)
-				qdel(pick(C.contents))
+	generateManifest(C, paying_account, pack)
 	return C
 
-//Paperwork for NT
-/obj/item/folder/paperwork
```
Comment (Trilbyspaceclone) [inline code/modules/cargo/order.dm:104]: 0 Idea why your so insistant on removing everything I add into cargo.

**Comment by Trilbyspaceclone** [comment]
Also 0 idea why your randomly removing paperwork stuff

File: code/modules/cargo/packs/goodies.dm [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	cost = 800
+	contains = list(/obj/item/clothing/gloves/color/yellow)
+
+/datum/supply_pack/goody/firstaidbruises_single
+	name = "Bruise Treatment Kit Single-Pack"
+	desc = "A single brute first-aid kit, perfect for recovering from being crushed in an airlock. Did you know people get crushed in airlocks all the time? Interesting..."
+	cost = 330
+	contains = list(/obj/item/storage/firstaid/brute)
+
+/datum/supply_pack/goody/firstaidburns_single
```
Comment (Trilbyspaceclone) [inline code/modules/cargo/packs/goodies.dm:25]: Why cant cargo buy any more firstaid packs of any type? Thats really limiting on what cargo can restock : /

File: code/modules/cargo/order.dm (hunk 2 above)
Comment (Ghommie) [reply code/modules/cargo/order.dm:104]: paperwork was already dead, I just forgot to remove these typepaths.

File: code/modules/cargo/packs/goodies.dm (hunk 3 above)
Comment (Ghommie) [reply code/modules/cargo/packs/goodies.dm:25]: medbay is stuffed full of medikits anyway. Why care about a non-issue?

**Comment by Trilbyspaceclone** [comment]
I also fully hate the idea of making medical kits non-buyable via cargo packets and forcing you to privetly buy them to restock medical

**Pull request merged**